phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -1714,9 +1716,14 @@ public void createTable(RpcController controller, CreateTableRequest request,
             }
 
             // check if the table was dropped, but had child views that were have not yet been cleaned up
-            if (!Bytes.toString(schemaName).equals(QueryConstants.SYSTEM_SCHEMA_NAME) &&
-                    clientVersion >= MIN_SPLITTABLE_SYSTEM_CATALOG) {
-                ViewUtil.dropChildViews(env, tenantIdBytes, schemaName, tableName);
+            if (!Bytes.toString(schemaName).equals(QueryConstants.SYSTEM_SCHEMA_NAME)) {
+                byte[] sysCatOrSysChildLink = SchemaUtil.getPhysicalTableName(
+                        clientVersion >= MIN_SPLITTABLE_SYSTEM_CATALOG ?
```

Comment:
nit: this would be much easier to read if you extracted the clientVersion -> table byte logic into its own method

Reply:
I did this as part of PHOENIX-5544 (https://github.com/apache/phoenix/pull/620)